controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,201 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"golang.org/x/exp/maps"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	"github.com/openshift/oadp-operator/pkg/common"
+)
+
+const (
+	nonAdminPrefix            = "non-admin-"
+	controllerManager         = "controller-manager"
+	containerName             = nonAdminPrefix + "manager"
+	nonAdminControllerManager = common.OADPOperatorPrefix + nonAdminPrefix + controllerManager
+	controlPlaneKey           = "control-plane"
+)
+
+var (
+	controlPlaneLabel = map[string]string{
+		controlPlaneKey: nonAdminPrefix + controllerManager,
+	}
+	deploymentLabels = map[string]string{
+		"app.kubernetes.io/component":  "manager",
+		"app.kubernetes.io/created-by": common.OADPOperator,
+		"app.kubernetes.io/instance":   nonAdminPrefix + controllerManager,
+		"app.kubernetes.io/managed-by": "kustomize",
+		"app.kubernetes.io/name":       "deployment",
+		"app.kubernetes.io/part-of":    common.OADPOperator,
```

Comment:
Should we set `OwnerReference` from OADP to NAC ?

Reply:
this is enough https://github.com/openshift/oadp-operator/pull/1370/files/c77bb834da638b56d2942795b25c83420e34ad01#diff-ece703e5852171fe350087ab8bd7d5e14c7b577a83664fadc7506cafd54117e2R108 ?

Reply:
Thanks